client/my-sites/plans/jetpack-plans/use-get-tiered-products.ts

```diff
@@ -0,0 +1,149 @@
+import {
+	PRODUCT_JETPACK_BACKUP_T1_YEARLY,
+	PRODUCT_JETPACK_BACKUP_T1_MONTHLY,
+	PRODUCT_JETPACK_BACKUP_T2_YEARLY,
+	PRODUCT_JETPACK_BACKUP_T2_MONTHLY,
+	PLAN_JETPACK_SECURITY_T1_YEARLY,
+	PLAN_JETPACK_SECURITY_T1_MONTHLY,
+	PLAN_JETPACK_SECURITY_T2_YEARLY,
+	PLAN_JETPACK_SECURITY_T2_MONTHLY,
+	TERM_ANNUALLY,
+	TERM_MONTHLY,
+	getJetpackStorageAmountDisplays,
+	JetpackSlugsWithStorage,
+} from '@automattic/calypso-products';
+import { translate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import { getSitePurchases } from 'calypso/state/purchases/selectors';
+import getSitePlan from 'calypso/state/sites/selectors/get-site-plan';
+import getSiteProducts from 'calypso/state/sites/selectors/get-site-products';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import slugToSelectorProduct from './slug-to-selector-product';
+import { Duration, SelectorProduct } from './types';
+
+const getProductWithOverrides = ( slug: JetpackSlugsWithStorage ) => {
+	const jetpackStorageAmountDisplays = getJetpackStorageAmountDisplays();
+
+	return {
+		...slugToSelectorProduct( slug ),
+		displayName: jetpackStorageAmountDisplays[ slug ],
+		buttonLabel: translate( 'Upgrade storage' ),
+		description: null,
+		subheader: translate( 'of backup storage', {
+			comment: 'A subheader that together with the header will read like "20GB of backup storage"',
+		} ),
+	};
+};
+
+export const useGetTieredBackupProducts = ( billingPeriod: Duration ) => {
+	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+
+	const activeProductSlugs =
+		useSelector( ( state ) => getSiteProducts( state, siteId ) )
+			?.filter( ( product ) => ! product.expired )
+			?.map( ( { productSlug } ) => productSlug ) ?? [];
+
+	const purchasedTier1BackupProduct = activeProductSlugs
+		?.filter( ( productSlug ) =>
+			[ PRODUCT_JETPACK_BACKUP_T1_YEARLY, PRODUCT_JETPACK_BACKUP_T1_MONTHLY ].includes(
+				productSlug
+			)
+		)
+		?.pop() as JetpackSlugsWithStorage;
```

Comment:
Non-blocking, but now that we're coercing to an empty array, the safe method calls here are unnecessary:

```suggestion
		.filter( ( productSlug ) =>
			[ PRODUCT_JETPACK_BACKUP_T1_YEARLY, PRODUCT_JETPACK_BACKUP_T1_MONTHLY ].includes(
				productSlug
			)
		)
		.pop() as JetpackSlugsWithStorage;
```